### PR TITLE
Fixed typos and incrementally improving documentation

### PR DESCRIPTION
**Pull request by kaeawc**
* Ran the project through Android Studio's code analyzer for typos. Tried to make sure I was only renaming internal variables / classes if it was code so as to not make any public API changes.
* Also started filling in some of the missing description fields for params and returns, or deleting @return when a function was a void type.

### CODE REVIEW

File: constraintlayout/core/src/test/java/androidx/constraintlayout/core/LinearEquation.java [hunk 1]
```diff
@@ -16,15 +16,19 @@
...
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 
 /**
  * LinearEquation is used to represent the linear equations fed into the solver.<br>
- * A linear equation can be an equality or an inequation (left term &le; or &ge; to the right term).<br>
- * The general form will be similar to a0x0 + a1x1 + ... = C + a2x2 + a3x3 + ... , where a0x0 is a term representing
- * a variable x0 of an amount a0, and C represent a constant term. The amount of terms on the left side or the right
+ * A linear equation can be an equality or an inequality (left term &le; or &ge; to the right term).<br>
```
Comment (camaelon) [inline constraintlayout/core/src/test/java/androidx/constraintlayout/core/LinearEquation.java:29]: inequation is correct actually

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionLayout.java [hunk 2]
```diff
@@ -2170,11 +2199,11 @@ public void getViewVelocity(View view, float posOnViewX, float posOnViewY, float
         if (mInterpolator != null) {
             float deltaT = EPSILON;
             float dir = Math.signum(mTransitionGoalPosition - mTransitionLastPosition);
-            float interpos = mInterpolator.getInterpolation(mTransitionLastPosition + deltaT);
+            float interpolation = mInterpolator.getInterpolation(mTransitionLastPosition + deltaT);
```
Comment (camaelon) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionLayout.java:2202]: interpolatedPosition ?

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ public void setStringValue(String value) {
      *
      * @return Typically 1 but 3 for colors.
      */
-    public int noOfInterpValues() {
+    public int numberOfInterpolationValues() {
```
Comment (camaelon) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java:106]: numberOfInterpolatedValues ?

File: constraintlayout/core/src/test/java/androidx/constraintlayout/core/LinearEquation.java (hunk 1 above)
Comment (kaeawc) [reply constraintlayout/core/src/test/java/androidx/constraintlayout/core/LinearEquation.java:29]: Ah okay, I'll check it into the project dictionary :)

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionLayout.java (hunk 2 above)
Comment (kaeawc) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionLayout.java:2202]: Will fix

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java (hunk 3 above)
Comment (kaeawc) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java:106]: Yeah that sounds better, will fix 👍🏻

Comment (kaeawc) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java:106]: @camaelon the comment above in the docs says `The number of interpolation values that need to be interpolated` - should that be left alone? I think that was why I renamed it that way originally.

**Pull request merged**